### DISCUSSION TITLE
Attaching a GPX to a road: only recorded tracks

### DISCUSSION THREAD

**Discussion by barteis** [osmandapp/OsmAnd#14929] [Q&A]
v 4.2.7 2022-7-13

Is it now the case that only recorded tracks can be attached to roads? (Or was it always so?) In attempting to do that with a GPX from AllTrails, that's the error message I see. 

It's a matter of interest to me because the instructional video I created on using dowloaded GPXs from the web may turn out to be incorrect.  
https://vimeo.com/654264099

Thanks.

![Screenshot_20220725-082159](https://user-images.githubusercontent.com/36173950/180817872-7a9ce7c6-838e-4013-b711-f75635d75c78.png)

**Comment by LancashireLad** [comment]
I have the same issue that appeared on the latest update to 4.2.7 released 2022-07-13. Prior to this it worked with perfectly with imported GPX tracks from https://cycle.travel/ (eg https://cycle.travel/map/journey/343471). May be there is something in the downloaded GPX that the app doesn't recognise?

**Reply by barteis** [reply]
That could well be the case. I looked at the code for two GPX tracks, one from GraphHopper, which Snapped to the road, the other from AllTrails, which got the error message as seen in my screenshot.  There were small differences in the line spacing (would that matter?) and in the  header tags. 

(Downloading a _track_ from GraphHopper produced a GPX with the street names included.  OsmAnd seemed to be able to navigate it as is, showing turn-by-turn directions.)

**Reply by barteis** [reply]
#14946

**Reply by vshcherb** [reply]
Indeed some investigation is going on ) https://github.com/osmandapp/OsmAnd/issues/14946

**Reply by vshcherb** [reply]
We want to avoid attaching weird tracks so we probably pushed too far by allowing to attach only recorded tracks. We want to avoid people to attach to roads random street lines cause it doesn't work simply.

**Reply by barteis** [reply]
I can see how that could be a problem.  Thanks for looking into!

**Comment by LancashireLad** [comment]
Quite by chance I seem to have found a work around...

My Places - tracks (select imported GPX) - track - altitude -calculate altitude - attach to roads - continue - choose profile - apply (when it eventually becomes available - takes a while) - Done (top right) Bingo!
